exporter/fileexporter/file_exporter.go

```diff
@@ -27,20 +29,27 @@ import (
 	"gopkg.in/natefinch/lumberjack.v2"
 )
 
-// Marshaler configuration used for marhsaling Protobuf to JSON.
-var tracesMarshaler = ptrace.NewJSONMarshaler()
-var metricsMarshaler = pmetric.NewJSONMarshaler()
-var logsMarshaler = plog.NewJSONMarshaler()
+const (
+	marshalTypeProto    = "proto"
+	marshalTypeProtobuf = "protobuf"
+)
 
 // fileExporter is the implementation of file exporter that writes telemetry data to a file
-// in Protobuf-JSON format.
 type fileExporter struct {
 	path  string
 	file  io.WriteCloser
 	mutex sync.Mutex
+
+	// isJSON defines whether the exported data is in json format
+	isJSON bool
+
+	tracesMarshaler  ptrace.Marshaler
+	metricsMarshaler pmetric.Marshaler
+	logsMarshaler    plog.Marshaler
 }
 
 func newFileExporter(conf *Config) *fileExporter {
+	tracesMarshaler, metricsMarshaler, logsMarshaler := buildMarshaler(conf.MarshalType)
```

Comment:
I think we should not always create marshalers for all signal types. Instead, we should look for a way to use `createTracesExporter`, `createMetricsExporter`, and `createLogsExporter` to indicate which marshalers are required.